src/QuartoPlayerAgent.java

```diff
@@ -146,28 +130,18 @@ protected String moveSelectionAlgorithm(int pieceID)
         QuartoPiece givenPiece = this.curState.board.getPiece(pieceID);
         QuartoGameState prevState = this.curState;
 
-        // Set this.curState
-        QuartoGameTransition quartoGameTransition = this.curState.transitions.get(givenPiece.binaryStringRepresentation() + ":" +
-                                                    minisChosenSquare[0] + "," + minisChosenSquare[1]);
+        QuartoGameTransition quartoGameTransition = prevState.transitions.get(prevState.bestTransition.placedPiece.binaryStringRepresentation() + ":" +
+                                                    minisChosenSquare[0] + "," + minisChosenSquare[1] + ":" + givenPiece.binaryStringRepresentation());
 
         if(quartoGameTransition == null) {
-            ArrayList<int[]> tempSquares = new ArrayList<int[]>();
-            for (int[] square: prevState.freeSquares) {
-                if (square[0] != minisChosenSquare[0] || square[1] != minisChosenSquare[1]) {
-                    tempSquares.add(square.clone());
-                }
-            }
-            prevState.freeSquares = tempSquares;
-            prevState.freePieces.remove(givenPiece);
-            this.curState = new QuartoGameState(prevState.board, prevState.freeSquares, prevState.freePieces,
-                                                prevState.alpha, prevState.beta, !prevState.isMaxState);
+            this.curState = new QuartoGameState(this.quartoBoard, Integer.MIN_VALUE, Integer.MAX_VALUE, true);
```

Comment:
Pretty sure that the state we'd be switching into should be a mini state, we've just made our decisions and now we're waiting for mini to make hers.